File: uf-client-service/src/main/java/com/kynetics/uf/android/ui/fragment/AuthorizationDialogFragment.java
/*
 * Copyright © 2017-2018  Kynetics  LLC
 *
 *  All rights reserved. This program and the accompanying materials
 *  are made available under the terms of the Eclipse Public License v1.0
 *  which accompanies this distribution, and is available at
 *  http://www.eclipse.org/legal/epl-v10.html
 */

package com.kynetics.uf.android.ui.fragment;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v4.app.Fragment;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * A simple {@link Fragment} subclass.
 * Activities that contain this fragment must implement the
 * {@link AuthorizationDialogFragment.OnAuthorization} interface
 * to handle interaction events.
 * Use the {@link AuthorizationDialogFragment#newInstance} factory method to
 * create an instance of this fragment.
 *
 * @author Daniele Sergio
 */
public class AuthorizationDialogFragment extends DialogFragment {

    public interface OnAuthorization {
        void onAuthorizationGrant();
        void onAuthorizationDenied();
    }

    public AuthorizationDialogFragment() {
        // Required empty public constructor
    }

    private static final String ARG_DESCRIPTION = "description_arg";
    private static final String ARG_TITLE = "title_arg";
    private static final String ARG_POSITIVE_BUTTON = "positive_button_arg";
    private static final String ARG_NEGATIVE_BUTTON = "negative_button_arg";

    private String mDescription, mTitle, mPositiveButton, mNegativeButton;


    public static AuthorizationDialogFragment newInstance(String title, String description, String positiveButton, String negativeButton) {
        AuthorizationDialogFragment fragment = new AuthorizationDialogFragment();
        Bundle args = new Bundle();
        args.putString(ARG_TITLE, title);
        args.putString(ARG_DESCRIPTION, description);
        args.putString(ARG_POSITIVE_BUTTON, positiveButton);
        args.putString(ARG_NEGATIVE_BUTTON, negativeButton);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mTitle = getArguments().getString(ARG_TITLE);
            mDescription = getArguments().getString(ARG_DESCRIPTION);
            mPositiveButton = getArguments().getString(ARG_POSITIVE_BUTTON);
            mNegativeButton = getArguments().getString(ARG_NEGATIVE_BUTTON);
        }
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return new AlertDialog.Builder(getActivity())
                .setTitle(mTitle)
                .setMessage(mDescription)
                .setPositiveButton(mPositiveButton,
                        (dialog, whichButton) -> onAuthorizationGrant()
                )
                .setNegativeButton(mNegativeButton,
                        (dialog, whichButton) -> onAuthorizationDenied()
                )
                .create();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view =  super.onCreateView(inflater, container, savedInstanceState);
        setCancelable(false);
        return view;
    }

    private void onAuthorizationGrant(){
        mListener.onAuthorizationGrant();
    }

    private void onAuthorizationDenied(){
        mListener.onAuthorizationDenied();
    }



    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnAuthorization) {
            mListener = (OnAuthorization) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    private OnAuthorization mListener;

}

File: uf-client-service/src/main/kotlin/com/kynetics/uf/android/ui/fragment/AuthorizationDialogFragment.kt
/*
 * Copyright © 2017-2020  Kynetics  LLC
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 */
package com.kynetics.uf.android.ui.fragment

import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment

/**
 * Activities that contain this fragment must implement the
 * [AuthorizationDialogFragment.OnAuthorization] interface
 * to handle interaction events.
 * Use the [AuthorizationDialogFragment.newInstance] factory method to
 * create an instance of this fragment.
 *
 * @author Daniele Sergio
 */
class AuthorizationDialogFragment : DialogFragment() {
    interface OnAuthorization {
        fun onAuthorizationGrant()
        fun onAuthorizationDenied()
    }

    private var mDescription: String? = null
    private var mTitle: String? = null
    private var mPositiveButton: String? = null
    private var mNegativeButton: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            mTitle = arguments!!.getString(ARG_TITLE)
            mDescription = arguments!!.getString(ARG_DESCRIPTION)
            mPositiveButton = arguments!!.getString(ARG_POSITIVE_BUTTON)
            mNegativeButton = arguments!!.getString(ARG_NEGATIVE_BUTTON)
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(activity!!)
                .setTitle(mTitle)
                .setMessage(mDescription)
                .setPositiveButton(mPositiveButton
                ) { dialog: DialogInterface?, whichButton: Int -> onAuthorizationGrant() }
                .setNegativeButton(mNegativeButton
                ) { dialog: DialogInterface?, whichButton: Int -> onAuthorizationDenied() }
                .create()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = super.onCreateView(inflater, container, savedInstanceState)
        isCancelable = false
        return view
    }

    private fun onAuthorizationGrant() {
        mListener!!.onAuthorizationGrant()
    }

    private fun onAuthorizationDenied() {
        mListener!!.onAuthorizationDenied()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mListener = if (context is OnAuthorization) {
            context
        } else {
            throw RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener")
        }
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    private var mListener: OnAuthorization? = null

    companion object {
        private const val ARG_DESCRIPTION = "description_arg"
        private const val ARG_TITLE = "title_arg"
        private const val ARG_POSITIVE_BUTTON = "positive_button_arg"
        private const val ARG_NEGATIVE_BUTTON = "negative_button_arg"
        fun newInstance(title: String?, description: String?, positiveButton: String?, negativeButton: String?): AuthorizationDialogFragment {
            val fragment = AuthorizationDialogFragment()
            val args = Bundle()
            args.putString(ARG_TITLE, title)
            args.putString(ARG_DESCRIPTION, description)
            args.putString(ARG_POSITIVE_BUTTON, positiveButton)
            args.putString(ARG_NEGATIVE_BUTTON, negativeButton)
            fragment.arguments = args
            return fragment
        }
    }
}